src/Container/GenericContainer.php

```diff
@@ -12,21 +12,28 @@
 use ReflectionUnionType;
 use function Tempest\attribute;
 use Tempest\Container\Exceptions\CannotAutowireException;
-use Tempest\Container\Exceptions\InvalidInitializerException;
 use Throwable;
 
 final class GenericContainer implements Container
 {
     use HasInstance;
 
     public function __construct(
-        private array $definitions = [],
-        private array $singletons = [],
-        /** @var (Initializer&CanInitialize)[] */
-        private array $initializers = [],
-        private ContainerLog $log = new InMemoryContainerLog(),
-    ) {
-    }
+        public array $definitions = [],
+        public array $singletons = [],
+        /**
+         * @template T of \Tempest\Container\Initializer
+         * @var class-string<T> $initializers
+         */
+        public array $initializers = [],
+
+        /**
+         * @template T of \Tempest\Container\DynamicInitializer
+         * @var class-string<T> $dynamicInitializers
+         */
+        public array $dynamicInitializers = [],
+        private readonly ContainerLog $log = new InMemoryContainerLog(),
```

Comment:
Seems like we could clean this up. Not sure why we need to expose some of this stuff.